src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -93,11 +83,18 @@ internal async Task SearchAsync(CancellationToken cancellationToken)
                 using var navigateToSearch = Logger.LogBlock(FunctionId.NavigateTo_Search, KeyValueLogMessage.Create(LogType.UserAction), cancellationToken);
                 using var asyncToken = _asyncListener.BeginAsyncOperation(GetType() + ".Search");
 
-                // We consider ourselves fully loaded when both the project system has completed loaded us, and we've
-                // totally hydrated the oop side.  Until that happens, we'll attempt to return cached data from languages
-                // that support that.
-                isFullyLoaded = await _host.IsFullyLoadedAsync(cancellationToken).ConfigureAwait(false);
-                await SearchAllProjectsAsync(isFullyLoaded, cancellationToken).ConfigureAwait(false);
+                if (searchCurrentDocument)
```

Comment:
top level logic split between searching just hte current doc (which is a unique codepath) vs searching all the projects.